### DISCUSSION TITLE
[SWCK]Why swck listen pods update event

### DISCUSSION THREAD

**Discussion by lphu** [apache/skywalking#11191] [Q&A] [answered]
swck listened for the create and update events of PODs, but did not set idempotent in the **/mutate-v1-pod** interface, which resulted in some scenarios updating PODs (such as adding region/zone annotations), swck will repeatedly execute the injection process, causing PODs to fail to start normally.

Wanna to understand the necessity of listenning updating event , even if an update is made, it also need to restart the Java process to make it attach again.

**Comment by wu-sheng** [comment]
The behaviors of the agent are defined through the values of system env variables, I believe watching the events makes sense.

**Reply by lphu** [reply]
Yes, injecting environment variables makes sense. The repeated injection contains env variables and the VolumeMount of the pod, [SWCK](https://github.com/apache/skywalking-swck/blob/master/operator/pkg/operator/injector/injector.go#L92) just appended the VolumeMount while pods updating,  then the starting pod will throw **Duplicate volumeMount**

Is this a swck bug or is there a problem with my usage?

**Reply by wu-sheng** [reply]
@innerpeacez @dashanji Could you reccheck?

**Reply by dashanji** [reply]
@lphu Thanks for reporting the problem. Actually, the current design is only for the **Create** event. As you said, the permission and the logic of the webhook are not idempotent. We could delete the **Update** event simply or achieve the update logic in the webhook. 

BTW, Could you please clarify why need to update the injected pod rather than create a new one?

**Reply by lphu** [reply]
Sometimes we may patch annotations such as Region or Instance ID after the Pod scheduling is completed. You can see example here. [Ali ECI](https://help.aliyun.com/document_detail/144561.html?spm=a2c4g.209123.0.0.126f45e3kmUlnq)

**Reply by dashanji** [reply, marked as answer]
@lphu I see, thanks for the details. Would you want to propose a PR to achieve the update logic in the webhook?
